### PR TITLE
cython3 compatibility

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
Change source so it compiles on 0.29.x and 3.0 (beta).

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
@mgorny can you review?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -30,9 +30,14 @@ jobs:
           sudo apt-get install -y libfuse-dev
 
       - name: Install Python dependencies
-        run: |
-          test/ci-install.sh
+        if: ${{ matrix.python-version != '3.11' }}
+        run: pip install pytest "sphinx<7.0" "Cython<3"
+
+      - name: Install Python dependencies
+        if: ${{ matrix.python-version == '3.11' }}
```
Comment (mgorny) [inline .github/workflows/test.yml:37]: Perhaps make Cython version another variable in the matrix, and test at least one Cython version with all the Pythons. Not saying that's likely to discover anything new but it should make it clearer whether the issue stems from Python 3.11 or Cython 3.

File: src/fuse_api.pxi [hunk 2]
```diff
@@ -387,7 +387,7 @@ cdef void* worker_start(void* data) with gil:
         session_loop(wd.buf, wd.bufsize)
     except:
         fuse_session_exit(session)
-        tid = wd.thread_id
+        tid = <uintptr_t> wd.thread_id
```
Comment (mgorny) [inline src/fuse_api.pxi:390]: For the record, I don't think there's a guarantee that `pthread_t` will be cleanly castable to an integer but I suppose there's no harm doing that until someone actually tries to use with with non-Linux FUSE implementation and complains ;-).

**Review by mgorny — APPROVED** [review]
Thanks! I can confirm that this fixes the build for me.

File: src/fuse_api.pxi (hunk 2 above)
Comment (ThomasWaldmann) [reply src/fuse_api.pxi:390]: Is there a better way to implement this and avoid the compiler warning? tid is only used for log output.

File: .github/workflows/test.yml (hunk 1 above)
Comment (ThomasWaldmann) [reply .github/workflows/test.yml]: done.

File: src/fuse_api.pxi (hunk 2 above)
Comment (mgorny) [reply src/fuse_api.pxi:390]: I'm afraid this is a non-trivial problem. Roughly, you could:

1. "dump" `pthread_t` but it's probably not worth the complexity for a debug print (basically print it as hex byte-by-byte)
2. use non-portable getter inside the thread to get its numeric TID and pass it but it's complex and then you'd need extra compatibility if different platforms are involved, see e.g. https://github.com/llvm/llvm-project/blob/llvmorg-16.0.3/llvm/lib/Support/Unix/Threading.inc#L113-L134
3. just create your own thread numbers and assign them consecutively (e.g. `thread 1`, `thread 2` and so on)

See e.g. https://stackoverflow.com/questions/1759794/how-to-print-pthread-t

Comment (ThomasWaldmann) [reply src/fuse_api.pxi:390]: thanks for the info, I created https://github.com/python-llfuse/python-llfuse/issues/85 so we can improve this later.

**Comment by mgorny** [comment]
If that wouldn't be a problem, could you make a new release, please?

**Comment by ThomasWaldmann** [comment]
@mgorny yes, I'll release 1.4.4 soon.

**Comment by ThomasWaldmann** [comment]
@mgorny 1.4.4 is released, see pypi and github releases.

**Comment by mgorny** [comment]
Thanks a lot! I can confirm that it works for us.

**Pull request merged**